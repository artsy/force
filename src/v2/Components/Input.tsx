import { Box, color, space } from "@artsy/palette"
import { growAndFadeIn } from "v2/Assets/Animations"
import { garamond, unica } from "v2/Assets/Fonts"
import * as React from "react";
import styled from "styled-components"
import { block } from "./Helpers"
import { borderedInputMixin } from "./Mixins"

export interface InputProps extends React.HTMLProps<HTMLInputElement> {
  block?: boolean
  description?: string
  error?: string
  ref?: React.RefObject<any>
  title?: string
}

/**
 * @deprecated in favor of our Design System Input component in @artsy/palette
 * https://palette.artsy.net/elements/inputs/input
 */
export const Input: React.ExoticComponent<InputProps> = React.forwardRef(
  ({ error, title, description, ...rest }, ref) => {
    return (
      <Box>
        {title && <Title>{title}</Title>}
        {description && <Description>{description}</Description>}
        <StyledInput ref={ref} hasError={!!error} {...rest} />
        {error && <InputError>{error}</InputError>}
      </Box>
    )
  }
)

export const StyledInput = styled.input`
  ${borderedInputMixin};
  ${block(24)};

  /* Remove shadow and default border radius on iOS */
  background-clip: padding-box;
  border-radius: 0;
`

export const Title = styled.div`
  ${garamond("s17")};
`

const Description = styled.div`
  ${garamond("s15")};
  color: ${color("black60")};
  margin: ${space(0.3)}px 0 0;
`

export const InputError = styled.div`
  ${unica("s12")};
  margin-top: ${space(1)}px;
  color: ${color("red100")};
  transition: visibility 0.2s linear;
  animation: ${growAndFadeIn("16px")} 0.25s linear;
  height: 16px;
`

export default Input
