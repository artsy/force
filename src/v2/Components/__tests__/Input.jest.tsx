import { mount } from "enzyme"
import "jest-styled-components"
import renderer from "react-test-renderer"

import Input from "../Input"

describe("Input", () => {
  it("renders an input with no metadata", () => {
    const wrapper = mount(<Input />)

    expect(wrapper.find("input").length).toEqual(1)
  })

  it("passes supported props through to input", () => {
    const wrapper = mount(<Input type="password" placeholder="a placeholder" />)

    const inputProps = wrapper.find("input").props()

    expect(inputProps.type).toEqual("password")
    expect(inputProps.placeholder).toEqual("a placeholder")
  })

  it("renders the input metadata", () => {
    const wrapper = mount(
      <Input
        title="hello"
        description="This is a field"
        error="But something went wrong!"
      />
    )

    const actual = wrapper.text()
    expect(actual).toContain("hello")
    expect(actual).toContain("This is a field")
    expect(actual).toContain("But something went wrong!")
  })

  it("renders as a snapshot", () => {
    const component = renderer
      .create(
        <Input
          type="password"
          placeholder="a placeholder"
          title="hello"
          description="This is a field"
          error="But something went wrong!"
        />
      )
      .toJSON()
    expect(component).toMatchSnapshot()
  })
})
