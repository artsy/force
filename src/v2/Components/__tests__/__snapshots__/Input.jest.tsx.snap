// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input renders as a snapshot 1`] = `
.c2 {
  padding: 10px;
  box-shadow: none;
  -webkit-transition: border-color 0.25s;
  transition: border-color 0.25s;
  margin-right: 10px;
  resize: none;
  display: block;
  overflow: hidden;
  text-overflow: ellipsis;
  white-space: nowrap;
  font-family: 'Adobe Garamond W08','adobe-garamond-pro','AGaramondPro-Regular','Times New Roman',Times,serif;
  font-size: 17px;
  line-height: 1.1em;
  -webkit-font-smoothing: antialiased;
  border: 1px solid #f7625a;
  -webkit-transition: border-color 0.25s;
  transition: border-color 0.25s;
  background-clip: padding-box;
  border-radius: 0;
}

.c2::-webkit-input-placeholder {
  color: #cccccc;
  text-overflow: ellipsis;
  line-height: normal;
}

.c2::-moz-placeholder {
  color: #cccccc;
  text-overflow: ellipsis;
  line-height: normal;
}

.c2:-ms-input-placeholder {
  color: #cccccc;
  text-overflow: ellipsis;
  line-height: normal;
}

.c2::placeholder {
  color: #cccccc;
  text-overflow: ellipsis;
  line-height: normal;
}

.c2:hover,
.c2:focus,
.c2.focused {
  border-color: #f7625a;
  outline: 0;
}

.c2:disabled {
  border: 2px dotted #e5e5e5;
}

.c0 {
  font-family: 'Adobe Garamond W08','adobe-garamond-pro','AGaramondPro-Regular','Times New Roman',Times,serif;
  font-size: 17px;
  line-height: 1.1em;
  -webkit-font-smoothing: antialiased;
}

.c1 {
  font-family: 'Adobe Garamond W08','adobe-garamond-pro','AGaramondPro-Regular','Times New Roman',Times,serif;
  font-size: 15px;
  line-height: 1.25em;
  -webkit-font-smoothing: antialiased;
  color: #666;
  margin: 3px 0 0;
}

.c3 {
  font-family: Unica77LLWebRegular,Arial,serif;
  -webkit-font-smoothing: antialiased;
  font-size: 12px;
  line-height: 1.4em;
  margin-top: 10px;
  color: #E82F1D;
  -webkit-transition: visibility 0.2s linear;
  transition: visibility 0.2s linear;
  -webkit-animation: hqnQVQ 0.25s linear;
  animation: hqnQVQ 0.25s linear;
  height: 16px;
}

<div
  className="Box-sc-15se88d-0"
>
  <div
    className="c0"
  >
    hello
  </div>
  <div
    className="c1"
  >
    This is a field
  </div>
  <input
    className="c2"
    placeholder="a placeholder"
    type="password"
  />
  <div
    className="c3"
  >
    But something went wrong!
  </div>
</div>
`;
